Source/WTF/wtf/ThreadSafeWeakHashSet.h

```diff
@@ -177,23 +179,23 @@ class ThreadSafeWeakHashSet final {
         Locker otherLocker { other.m_lock };
         m_map = std::exchange(other.m_map, { });
         m_operationCountSinceLastCleanup = std::exchange(other.m_operationCountSinceLastCleanup, 0);
+        m_maxOperationCountWithoutCleanup = std::exchange(other.m_maxOperationCountWithoutCleanup, 0);
     }
 
-    static constexpr unsigned initialMaxOperationCountWithoutCleanup = 512;
     ALWAYS_INLINE void amortizedCleanupIfNeeded() const WTF_REQUIRES_LOCK(m_lock)
     {
-        if (++m_operationCountSinceLastCleanup / 2 > m_map.size() || m_operationCountSinceLastCleanup > m_maxOperationCountWithoutCleanup) {
-            bool didRemove = m_map.removeIf([] (auto& pair) {
+        if (++m_operationCountSinceLastCleanup > m_maxOperationCountWithoutCleanup) {
+            m_map.removeIf([] (auto& pair) {
                 return pair.value->objectHasStartedDeletion();
             });
-            m_maxOperationCountWithoutCleanup = didRemove ? std::max(initialMaxOperationCountWithoutCleanup, m_maxOperationCountWithoutCleanup / 2) : m_maxOperationCountWithoutCleanup * 2;
             m_operationCountSinceLastCleanup = 0;
+            m_maxOperationCountWithoutCleanup = m_map.size() * 2;
```

Comment:
Should we protect against overflows and use std::numeric_limits<unsigned>::max() when `m_map.size() >= std::numeric_limits<unsigned>::max() / 2` ?

Otherwise we'll wrap against when the size becomes to large and start using small values for `m_maxOperationCountWithoutCleanup` again.